kernel/src/vm/pmm.c

```diff
@@ -46,19 +49,39 @@ void pmm_init(size_t start, size_t end) {
     size_t infoindex = 1;
     union MemorySlice * currentslice = firstslice;
 
+    size_t peripherals_base;
 
+    switch (get_hardwareinfo()->boardType) {
+        case RaspBerryPiTwo:
+            peripherals_base = BCM2836_PERIPHERALS_PHYSICAL_BASE;
+            break;
+        default:
+            FATAL("Board not implemented");
+    }
+    
     // Go through all memory and index it with sliceinfo structs.
     // We start 1 Memoryslice after `start` because we already allocated the first one at the start
     // of this function.
     for (union MemorySlice * i = ((union MemorySlice *)start + 1); i < ((union MemorySlice *)end);
          i++) {
         struct MemorySliceInfo * currentsliceinfo = &currentslice->bucketinfo[infoindex];
 
+        
+
+
         // make this sliceinfo struct point to the right slice
         currentsliceinfo->slice = i;
 
-        // Add the sliceinfo to the unused list
-        push_to_ll(&physicalMemoryManager.unused, currentsliceinfo);
+	// Add 21 MiB because RPi 2 & Zero has regular  16 MiB peripherals region, but RPi 2 also has 4 MiB region after
+        if (((size_t)i - 0x80000000) >= peripherals_base &&
+            ((size_t)i - 0x80000000) < peripherals_base + Mebibyte*21) {
```

Comment:
It's not that nice to hardcode these sizes here. Would be nice if hardware info had a function you could just call with ' is this address in a reserved region?' and it returning a boolean